opal/corelib/string.rb

```diff
@@ -436,6 +436,7 @@ def gsub(pattern, replacement = undefined, &block)
       }
 
       var result = '', match_data = nil, index = 0, match, _replacement;
+      var onlyAnchor = /\/[\^\$]\//.test(pattern.toString())
```

Comment:
`pattern.source` seems the way to go:

![screen shot 2016-12-09 at 01 38 46](https://cloud.githubusercontent.com/assets/1051/21033425/601a70e6-bdb0-11e6-8d1f-d4f3bffad873.png)

probably with something like: 

```js
pattern.source === '^' || pattern.source === '$'
```